File: ess/src/app/service/role.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { baseUrl } from './helper';

@Injectable({
  providedIn: 'root'
})
export class RoleService {

  constructor(private http: HttpClient) { }

  public addRole(role:any){
    return this.http.post(`${baseUrl}/role/`,role);
  }
  public updateRole(role:any){
    return this.http.put(`${baseUrl}/role/`,role);
  }
  public getallRole(){
    return this.http.get(`${baseUrl}/role/roles`);
  }
  public getRoleCount(){
    return this.http.get(`${baseUrl}/role/roles/count`);
  }
  public getRolebyid(id: Number){
    return this.http.get(`${baseUrl}/role/roles/${id}`);
  }
  public deleteRole(id: string){
    return this.http.delete(`${baseUrl}/role/delete/`+id);
  }

  

  


  
  

  
  

File: java/ess/src/main/java/com/ess/service/RoleService.java
package com.ess.service;


import com.ess.common.util.GenericResponse;
import com.ess.dto.Role;

public interface RoleService {
	
	public GenericResponse createRole(Role role) throws Throwable;
	public GenericResponse updateRole(Role role) throws Throwable;
	public GenericResponse getallRole();
	public GenericResponse getRoleCount();
	public GenericResponse deleteRole(long id);
	public GenericResponse getRolebyid(long id);

}
